lib/modules/manager/helmfile/artifacts.ts

```diff
@@ -0,0 +1,84 @@
+import is from '@sindresorhus/is';
+import { quote } from 'shlex';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions, ToolConstraint } from '../../../util/exec/types';
+import {
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import { getFile } from '../../../util/git';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.trace(`helmfile.updateArtifacts(${packageFileName})`);
+
+  const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+  if (
+    !isLockFileMaintenance &&
+    (updatedDeps === undefined || updatedDeps.length < 1)
+  ) {
+    logger.debug('No updated helmfile deps - returning null');
+    return null;
+  }
+
+  const lockFileName = getSiblingFileName(packageFileName, 'helmfile.lock');
+  const existingLockFileContent = await getFile(lockFileName);
+
+  if (is.falsy(existingLockFileContent)) {
+    logger.debug('No helmfile.lock found');
+    return null;
+  }
+
+  try {
+    await writeLocalFile(packageFileName, newPackageFileContent);
+    const helmfileToolConstraint: ToolConstraint = {
+      toolName: 'helmfile',
+      constraint: config.constraints?.helmfile,
```

Comment:
Please add to the docs that now `helmfile` supports constraints. https://github.com/renovatebot/renovate/blob/main/docs/usage/configuration-options.md#constraints

Reply:
Like this 55aa935 ?